submissions/iakow/friends-app/js/main.js

```diff
@@ -0,0 +1,232 @@
+"use strict";
+
+disableFilters();
+
+const USERS_AMOUNT = 100;
+
+mountUserList(getPlaceholders());
+
+const ALL_USER_CARDS = [];
+
+fetch(`https://randomuser.me/api/?results=${USERS_AMOUNT}`)
+  .then((response) => response.json())
+  .then((response) => {
+    response.results.forEach((userData) => {
+      ALL_USER_CARDS.push(getUserCard(userData));
+    });
+
+    mountUserList(ALL_USER_CARDS);
+    enableFilters();
+  });
+
+addHandlers();
+
+
+function addHandlers() {
+  const filtersBlock = document.querySelector("#filters-block");
+
+  filtersBlock.addEventListener("input", (e) => {
+    const { id } = e.target;
+
+    if (id === "filter-name") {
+      e.target.value = e.target.value.trim();
+    }
+
+    if (id === "filter-age-min" || id === "filter-age-max") {
+      if (e.target.value === "0") e.target.value = "";
+    }
+
+    applyFilters();
+  });
+
+  filtersBlock.addEventListener("click", (e) => {
+    const { id, value } = e.target;
+
+    if (id === "filter-sex") {
+      if (!value) {
+        e.target.value = "female";
+      } else if (value === "female") {
+        e.target.value = "male";
+      } else if (value === "male") {
+        e.target.value = "";
+      }
+    } else if (id === "sort-age") {
+      document.querySelector("#sort-name").value = "";
+
+      if (!value) {
+        e.target.value = "up";
+      } else if (value === "up") {
+        e.target.value = "down";
+      } else if (value === "down") {
+        e.target.value = "";
+      }
+    } else if (id === "sort-name") {
+      document.querySelector("#sort-age").value = "";
+
+      if (!value) {
+        e.target.value = "up";
+      } else if (value === "up") {
+        e.target.value = "down";
+      } else if (value === "down") {
+        e.target.value = "";
+      }
+    } else {
+      return;
+    }
+
+    applyFilters();
+  });
+
+  document.querySelector("#reset-filters").addEventListener("click", (e) => {
+    e.target.disabled = true;
+
+    document.querySelectorAll(".filter").forEach((ctrl) => (ctrl.value = ""));
+
+    mountUserList(ALL_USER_CARDS);
+  });
+}
+
+function applyFilters() {
+  const filteredArr = [...ALL_USER_CARDS]
+    .filter(filterByName)
+    .filter(filterByAge)
+    .filter(filterBySex)
+    .sort(sortByAge)
+    .sort(sortByName);
+
```

Comment:
It looks like it's not efficient to traverse the array 5 times on every event. But I did some research with a different version of the code like this
```javascript
let filteredArr = [...allCardElements];

  if (filterNameInpt.value) filteredArr = filterByName(filteredArr);
  if (filterAgeMinInpt.value || filterAgeMinInpt.value) filteredArr = filterByMinAge(filteredArr);
  if (filterGenderInpt.value) filteredArr = filterBySex(filteredArr);
  if (sortAgeInpt.value) sortByAge(filteredArr);
  if (sortNameInpt.value) sortByName(filteredArr);
```
and it turned out that the difference is about 2 milliseconds. So I decided to leave the cleaner and more understandable code, as it seems to me.